config/ogdl/src/main/java/org/apache/shiro/config/ogdl/ReflectionBuilder.java

```diff
@@ -287,10 +287,15 @@ private String parseBeanId(String lhs) {
             }
 
             processor.execute();
-        }
 
-        //SHIRO-413: init method must be called for constructed objects that are Initializable
-        LifecycleUtils.init(objects.values());
+            //SHIRO-778: onInit method on AuthenticatingRealm is called twice
+            objects.keySet().stream()
+                    .filter(key -> !kvPairs.containsKey(key))
+                    .forEach(key -> LifecycleUtils.init(objects.get(key)));
+        } else {
+            //SHIRO-413: init method must be called for constructed objects that are Initializable
+            LifecycleUtils.init(objects.values());
+        }
```

Comment:
This else branch will cause calling initialization witce.
Consider 
```
        builder.buildObjects(defs);
        builder.buildObjects(null);
```

Reply:
https://github.com/fpapon/shiro/pull/1

Reply:
I'm not sure that users can use the buildObjects, it's not a part of the public api so I can't see the case where this method will be call several time at startup.
But if we need to add a control, may be we can add a boolean method in the Initializable interface to define if the bean has been initialized.
@bdemers @bmhm thoughts?